clerk/verification.go

```diff
@@ -0,0 +1,72 @@
+package clerk
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+const (
+	CookieSession       = "__session"
+	QueryParamSessionId = "_clerk_session_id"
+	OriginHeader        = "Origin"
+)
+
+type VerificationService service
+
+type verifyRequest struct {
+	Token string `json:"token"`
+}
+
+func (s *VerificationService) Verify(req *http.Request) (*Session, error) {
```

Comment:
⛔ I'd suggest that we also export a pure plug n play, HTTP Go middleware too. For example

```go
func clerkMiddleware(next http.Handler) http.Handler {
  return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
    doClerkStuff()...
    next.ServeHTTP(w, r)  })
}
```

WDYT?

Reply:
Yes, I was planning to add something like that. 
I just wanted to first start using it, to make sure I implement it according to an actual use case. But I can also add it now and modify it accordingly once we include it in the DApi.

Reply:
👍